### PR TITLE
 pml_ucx: add ompi datatype attribute to release ucp_datatype - v4.0.x

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@brminich @hoopoepg can u pls review?

File: ompi/mca/pml/ucx/pml_ucx.c [hunk 1]
```diff
@@ -248,15 +253,34 @@ int mca_pml_ucx_init(void)
...
                     (void *)ompi_pml_ucx.ucp_context,
                     (void *)ompi_pml_ucx.ucp_worker);
-    return OMPI_SUCCESS;
+    return rc;
+
+err_destroy_worker:
+    ucp_worker_destroy(ompi_pml_ucx.ucp_worker);
+    ompi_pml_ucx.ucp_worker = NULL;
+err:
+    return OMPI_ERROR;
```
Comment (brminich) [inline ompi/mca/pml/ucx/pml_ucx.c]: need to return rc here as well, because you set it to different values in error cases

**Comment by hppritcha** [comment]
is this a fix for the massive memory leak reported on the ompi user list several days ago?

**Comment by yosefe** [comment]
@bwbarrett ok to merge?

**Pull request merged**